src/pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -243,7 +248,7 @@ def _reorder_callbacks(callbacks: List[Callback]) -> List[Callback]:
             A new list in which the last elements are Checkpoint if there were any present in the
             input.
         """
-        checkpoints = [c for c in callbacks if isinstance(c, Checkpoint)]
+        checkpoints: List[Callback] = [c for c in callbacks if isinstance(c, Checkpoint)]
```

Comment:
```suggestion
        checkpoints: List[Checkpoint] = [c for c in callbacks if isinstance(c, Checkpoint)]
```

Reply:
I saw an error when adding the two list together, mypy wants the types to match.